clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -2359,31 +2334,22 @@ public void testDeleteRecords() throws Exception {
                 assertTrue(e0.getCause() instanceof OffsetOutOfRangeException);
             }
 
-            // "leader not available" failure on metadata request for partition 2
+            // not authorized to delete records for partition 2
             KafkaFuture<DeletedRecords> myTopicPartition2Result = values.get(myTopicPartition2);
             try {
                 myTopicPartition2Result.get();
                 fail("get() should throw ExecutionException");
             } catch (ExecutionException e1) {
-                assertTrue(e1.getCause() instanceof LeaderNotAvailableException);
+                assertTrue(e1.getCause() instanceof TopicAuthorizationException);
             }
```

Comment:
This can be replaced with `assertThrows(TopicAuthorizationException.class, () ->myTopicPartition2Result.get(), "get() should throw ExecutionException" )`

Also, Should we change the comment into `TopicAuthorizationException`? It doesn't throw `ExecutionException` at all.

Same comment applies to other places.

Reply:
We are actually catching ExecutionException and checking the cause of the exception. This would not work because it's not throwing TopicAuthorizationException, but it is the supplied cause for the ExecutionException.

Reply:
Ah, I see. Thanks for the explanation.